serialization/src/main/scala/sbt/serialization/SerializedValue.scala

```diff
@@ -113,6 +131,10 @@ private final case class LazyValue[V](value: V, pickler: SPickler[V]) extends Se
   override def parse[T](implicit unpickler: Unpickler[T]): Try[T] =
     toJson.parse[T]
 
+  // as with parse, we should/could avoid JSON here
+  def hasTag[T](implicit unpickler: Unpickler[T]): Boolean =
+    toJson.hasTag[T]
```

Comment:
This should probably just be `pickler.tag == unpickler.tag`, rather than doing a pickle and unpickle.